lib/src/serialization_rule.dart

```diff
@@ -249,14 +249,15 @@ class MapRule extends SerializationRule {
   }
 
   void inflateNonEssentialFromList(List state, Map newMap, Reader r) {
+    bool isNextOneKey = true;
     var key;
     for (var each in state) {
-      if (key == null) {
+      if (isNextOneKey == true) {
```

Comment:
I just added it. I hope I did it right. It fails with the old version and succeeds with the new version in my local machine.